### PR TITLE
doc: intra-doc links and doc comments for build script

### PR DESCRIPTION
**Pull request by weihanglo**
### What does this PR try to resolve?

More docs and intra-doc links for `src/cargo/core/compiler/custom_build.rs`

### How should we test and review this PR?

```
cargo doc --document-private-items --no-deps --open
```

Then proofread it!

### CODE REVIEW

File: src/cargo/core/compiler/custom_build.rs [hunk 1]
```diff
@@ -1,3 +1,36 @@
+//! A module about how to execute a custom build script and parse its output.
```
Comment (epage) [inline src/cargo/core/compiler/custom_build.rs:1]: nit: "A module about" is a bit redundant, rustdoc makes it clear this is a module

Comment (epage) [inline src/cargo/core/compiler/custom_build.rs:1]: Side note: If `custom_build.rs` is really about build scripts, should we make that more obvious, whether thats renaming this to `build_script.rs` or `build_rs.rs`?

**Review by epage — APPROVED** [review]
Feel free to merge on my behalf whether you end up fixing it or not

File: src/cargo/core/compiler/custom_build.rs (hunk 1 above)
Comment (weihanglo) [reply src/cargo/core/compiler/custom_build.rs]: Had the same thought. I could see custom build/build script/build.rs co-exist everywhere. `build_script.rs` gets my vote and I can send a PR fixing all occurrences then.

**Pull request merged**